timemachine/cpp/src/kernels/k_indices.cu

```diff
@@ -0,0 +1,38 @@
+#include "k_indices.cuh"
+
+// Takes a source and destination array. Assumes K <= N with values in the src are less than or equal
+// to K. The value of the src is used as the indice and the value in the destination array. Allows combining
+// a series of indices to get a unique set of values.
```

Comment:
Might be worth documenting with equivalent numpy code. I think it'd be something like `dst[src] = src` in this case?